src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractTypeImportCompletionService.CacheEntry.cs

```diff
@@ -41,15 +45,19 @@ internal abstract partial class AbstractTypeImportCompletionService
                 string genericTypeSuffix,
                 bool isInternalsVisible,
                 bool isAttributeContext,
-                bool isCaseSensitive)
+                bool isCaseSensitive,
+                bool hideAdvancedMembers)
             {
                 // We will need to adjust some items if the request is made in:
                 // 1. attribute context, then we will not show or complete with "Attribute" suffix.
                 // 2. a project with different language than when the cache entry was created,
                 //    then we will change the generic suffix accordingly.
+                // 3. asked to hide advanced members and there is advanced member in the cache
                 // Otherwise, we can simply return cached items.
                 var isSameLanguage = Language == language;
-                if (isSameLanguage && !isAttributeContext)
+                if (isSameLanguage &&
+                    !isAttributeContext &&
+                    !(hideAdvancedMembers && ContainsAdvancedMembers))
```

Comment:
hoenstly, my brain sucks.  i'm having trouble understanding `&& !(hideAdvancedMembers && ContainsAdvancedMembers)`.  can you break this into a local that is easier to understand.